src/Aspire.Hosting/Publishing/ResourceContainerImageBuilder.cs

```diff
@@ -273,19 +273,21 @@ private async Task<bool> ExecuteDotnetPublishAsync(IResource resource, Container
 
             if (options.TargetPlatform is not null)
             {
-                // Use the appropriate MSBuild property based on the number of RIDs
+                // Use the appropriate MSBuild properties based on the number of RIDs
                 var runtimeIds = options.TargetPlatform.Value.ToMSBuildRuntimeIdentifierString();
                 var ridArray = runtimeIds.Split(';');
 
                 if (ridArray.Length == 1)
                 {
-                    // Single platform - use ContainerRuntimeIdentifier
+                    // Single platform - use RuntimeIdentifier/ContainerRuntimeIdentifier
+                    arguments += $" /p:RuntimeIdentifier=\"{ridArray[0]}\"";
                     arguments += $" /p:ContainerRuntimeIdentifier=\"{ridArray[0]}\"";
                 }
                 else
                 {
-                    // Multiple platforms - use RuntimeIdentifiers
+                    // Multiple platforms - use RuntimeIdentifiers/ContainerRuntimeIdentifiers
                     arguments += $" /p:RuntimeIdentifiers=\"{runtimeIds}\"";
```

Comment:
I'm hitting an issue when trying to use multiple RIDs:

```
❯ dotnet publish "D:\git\aspire2\playground\TestShop\CatalogDb\CatalogDb.csproj" --configuration Release /t:PublishContainer /p:ContainerRepository="catalogdbapp" /p:RuntimeIdentifiers="linux-x64;linux-arm64"
MSBUILD : error MSB1006: Property is not valid.
Switch: linux-arm64
```

@captainsafia - did you see this at all when you did multi arch support?

Reply:
I've pushed a change that should fix this.